packages/apps/graph/src/graph/objects/non-fungible-chain-account.ts

```diff
@@ -27,21 +30,43 @@ export default builder.node(
     },
     async loadOne({ chainId, accountName }) {
       try {
-        return {
-          __typename: NonFungibleChainAccountName,
+        return await getNonFungibleChainAccount({
           chainId,
           accountName,
-          nonFungibles: [],
-          transactions: [],
-        };
+        });
       } catch (error) {
         throw normalizeError(error);
       }
     },
     fields: (t) => ({
       chainId: t.exposeID('chainId'),
       accountName: t.exposeString('accountName'),
-      nonFungibles: t.field({
+      guard: t.field({
+        type: Guard,
+        complexity: COMPLEXITY.FIELD.CHAINWEB_NODE,
+        async resolve(parent) {
+          try {
+            const tokenDetails = await tokenDetailsLoader.load({
+              accountName: parent.accountName,
+              chainId: parent.chainId,
+            });
+
+            const accountDetails = await nonFungibleAccountDetailsLoader.load({
+              tokenId: tokenDetails[0].id,
```

Comment:
Can we make the assumption that for this account and chain the guard is the same for all tokens? @alber70g

Reply:
No we cannot :)
Basically each token has it's own tables. So also it's own accountnames, guards and balances. So they can have a different guard as well

Reply:
Feedback applied 👍